projects/gateway2/proxy_syncer/endpoints_prioritize.go

```diff
@@ -48,14 +53,29 @@ func NewPriorities(failoverPriority []string) *Prioritizer {
 	}
 }
 
-func (p *Prioritizer) GetPriority(proxyLabels, upstreamEndpointLabels map[string]string) int {
+func (p *Prioritizer) GetPriority(proxyLabels, upstreamEndpointLabels map[string]string, defaultNetwork string) int {
 	for j, label := range p.priorityLabels {
 		valueForProxy, ok := p.priorityLabelOverrides[label]
 		if !ok {
-			valueForProxy = proxyLabels[label]
+			valueForProxy, ok = proxyLabels[label]
```

Comment:
Could this logic be simplified? It feels like we perform some checks multiple times, when we could do them once.

- if the labels are defined for proxy and endpoint, and they differ, return the lowestPriority-j (ie exactly what existed before)
- if the label is not the istioLabel.TopologyNetwork.Name, continue
- if either of the labels are empty, continue

Reply:
just redid this to make it more readable